From 4a7eda24050b42072c77d527257609e743e8dc78 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Wed, 16 Nov 2011 20:09:40 +0000
Subject: Adapt to domain changes in videoabi 12

This is a blind change, I don't have this hardware handy to test with.
It probably wants the same new API that tdfx wants.

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
diff --git a/src/ast_driver.c b/src/ast_driver.c
index bd3d338..128538f 100644
--- a/src/ast_driver.c
+++ b/src/ast_driver.c
@@ -498,11 +498,20 @@ ASTPreInit(ScrnInfoPtr pScrn, int flags)
 	      (pScrn->chipset != NULL) ? pScrn->chipset : "Unknown ast");
 
    /* Resource Allocation */
+#if ABI_VIDEODRV_VERSION < 12
     pAST->IODBase = pScrn->domainIOBase;  
+#else
+    pAST->IODBase = 0;
+#endif
     /* "Patch" the PIOOffset inside vgaHW in order to force
      * the vgaHW module to use our relocated i/o ports.
      */
-    VGAHWPTR(pScrn)->PIOOffset = pAST->PIOOffset = pAST->IODBase + PCI_REGION_BASE(pAST->PciInfo, 2, REGION_IO) - 0x380;
+
+#if ABI_VIDEODRV_VERSION < 12
+    VGAHWPTR(pScrn)->PIOOffset = /* ... */
+#endif
+       	pAST->PIOOffset =
+	pAST->IODBase + PCI_REGION_BASE(pAST->PciInfo, 2, REGION_IO) - 0x380;
 	
     pAST->RelocateIO = (IOADDRESS)(PCI_REGION_BASE(pAST->PciInfo, 2, REGION_IO) + pAST->IODBase);
 	
--
cgit v0.9.0.2-2-gbebe
